src/app/options/options.component.ts

```diff
@@ -1,18 +1,44 @@
-import { Component, ViewChild, ElementRef } from '@angular/core';
+import { Component, ViewChild, ElementRef, OnInit } from '@angular/core';
 import { Note } from '@app/notes/notes.model';
 import { Filter, Options } from '@app/shared/model/options.model';
 import { NotesComponent } from '@app/notes/notes.component';
+import { Router, ActivatedRoute } from '@angular/router';
 
 @Component({
   selector: 'app-options',
   templateUrl: './options.component.html',
 })
-export class OptionsComponent {
+export class OptionsComponent implements OnInit {
   options: Options = new Options();
   filter: Filter = new Filter();
   @ViewChild(NotesComponent, { static: true }) noteChild;
 
-  constructor() {}
+  constructor(private router: Router, private route: ActivatedRoute) {}
+
+  ngOnInit() {
+    this.route.queryParamMap.subscribe(queryParamMap => {
+      /* tslint:disable:no-string-literal */
+      if ('filter' in this && this.filter.isEmpty()) {
+        for (const i of Object.keys(this.filter)) {
+          if (i in queryParamMap['params']) {
```

Comment:
Can we use something more idiomatic typescript here like:
```suggestion
          if (queryParamMap.get('params').includes(i)) {
```

This would also mean that we could get rid of the linter exclusion here.